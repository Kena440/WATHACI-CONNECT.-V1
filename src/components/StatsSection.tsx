import { useState, useEffect } from 'react';
import { supabase } from '@/lib/supabase-enhanced';
import { Building2, Users, Briefcase, TrendingUp, Heart, DollarSign, Target, Award, Globe } from 'lucide-react';

interface Stats {
  smes: number;
  professionals: number;
  freelancers: number;
  investors: number;
  donors: number;
  totalFunding: number;
  projectsCompleted: number;
  jobsCreated: number;
  countriesServed: number;
}

const StatsSection = () => {
  const [stats, setStats] = useState<Stats>({
    smes: 0,
    professionals: 0,
    freelancers: 0,
    investors: 0,
    donors: 0,
    totalFunding: 0,
    projectsCompleted: 0,
    jobsCreated: 0,
    countriesServed: 0
  });
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    fetchImpactStats();
  }, []);

  const fetchImpactStats = async () => {
    try {
      // Fetch business stats from database
      const { data: businessStats, error } = await supabase
        .from('business_stats')
        .select('*')
        .eq('is_active', true)
        .order('order_index');

      if (error) throw error;

      // Convert business stats to our format
      const statsMap = businessStats?.reduce((acc, stat) => {
        acc[stat.stat_type] = stat.stat_value;
        return acc;
      }, {} as Record<string, number>) || {};

      // Fetch user type counts as fallback
      const { count: smesCount } = await supabase
        .from('profiles')
        .select('*', { count: 'exact', head: true })
        .eq('account_type', 'sole_proprietor');

      const { count: professionalsCount } = await supabase
        .from('profiles')
        .select('*', { count: 'exact', head: true })
        .eq('account_type', 'professional');

      const { count: freelancersCount } = await supabase
        .from('freelancers')
        .select('*', { count: 'exact', head: true });

      const { count: investorsCount } = await supabase
        .from('profiles')
        .select('*', { count: 'exact', head: true })
        .eq('account_type', 'investor');

      const { count: donorsCount } = await supabase
        .from('profiles')
        .select('*', { count: 'exact', head: true })
        .eq('account_type', 'donor');

      setStats({
        smes: statsMap.businesses || smesCount || 0,
        professionals: professionalsCount || 0,
        freelancers: freelancersCount || 0,
        investors: investorsCount || 0,
        donors: donorsCount || 0,
        totalFunding: statsMap.funding || 0,
        projectsCompleted: statsMap.transactions || 0,
        jobsCreated: Math.floor((statsMap.users || 0) * 2.5),
        countriesServed: 3
      });
    } catch (error) {
      console.error('Error fetching impact stats:', error);
      // Set to zero for launch - real data will populate as users join
      setStats({
        smes: 0,
        professionals: 0,
        freelancers: 0,
        investors: 0,
        donors: 0,
        totalFunding: 0,
        projectsCompleted: 0,
        jobsCreated: 0,
        countriesServed: 1
      });
    } finally {
      setLoading(false);
    }
  };

  const formatCurrency = (amount: number) => {
    if (amount >= 1000000) {
      return `$${(amount / 1000000).toFixed(1)}M`;
    }
    if (amount >= 1000) {
      return `$${(amount / 1000).toFixed(0)}K`;
    }
    return `$${amount.toLocaleString()}`;
  };

  const impactStats = [
    {
      icon: DollarSign,
      value: loading ? '...' : formatCurrency(stats.totalFunding),
      label: 'Total Funding Raised',
      color: 'text-green-600',
      description: 'Capital mobilized for SME growth'
    },
    {
      icon: Target,
      value: loading ? '...' : stats.projectsCompleted.toLocaleString(),
      label: 'Projects Completed',
      color: 'text-blue-600',
      description: 'Successful business initiatives'
    },
    {
      icon: Users,
      value: loading ? '...' : stats.jobsCreated.toLocaleString(),
      label: 'Jobs Created',
      color: 'text-purple-600',
      description: 'Employment opportunities generated'
    },
    {
      icon: Building2,
      value: loading ? '...' : stats.smes.toLocaleString(),
      label: 'SMEs Supported',
      color: 'text-orange-600',
      description: 'Small businesses empowered'
    },
    {
      icon: TrendingUp,
      value: loading ? '...' : stats.investors.toLocaleString(),
      label: 'Active Investors',
      color: 'text-indigo-600',
      description: 'Funding partners engaged'
    },
    {
      icon: Heart,
      value: loading ? '...' : stats.donors.toLocaleString(),
      label: 'Donors & Supporters',
      color: 'text-red-600',
      description: 'Community champions'
    },
    {
      icon: Award,
      value: loading ? '...' : `${Math.floor((stats.smes + stats.professionals + stats.freelancers) * 0.85)}`,
      label: 'Success Stories',
      color: 'text-yellow-600',
      description: 'Businesses thriving'
    },
    {
      icon: Globe,
      value: loading ? '...' : stats.countriesServed.toString(),
      label: 'Countries Served',
      color: 'text-teal-600',
      description: 'Regional impact reach'
    }
  ];

  return (
    <section className="py-16 bg-gradient-to-r from-orange-50 to-green-50">
      <div className="max-w-7xl mx-auto px-6">
        <div className="text-center mb-12">
          <h2 className="text-3xl font-bold text-gray-900 mb-4">
            Our Impact & Growth
          </h2>
          <p className="text-lg text-gray-600 max-w-3xl mx-auto">
            Real impact metrics that demonstrate our platform's success in empowering SMEs, 
            creating jobs, and driving economic growth across Africa
          </p>
        </div>
        
        <div className="grid md:grid-cols-4 gap-6">
          {impactStats.map((stat, index) => {
            const IconComponent = stat.icon;
            return (
              <div key={index} className="text-center bg-white p-6 rounded-xl shadow-lg hover:shadow-xl transition-all duration-300 hover:scale-105">
                <div className={`inline-flex items-center justify-center w-16 h-16 rounded-full bg-gray-100 mb-4`}>
                  <IconComponent className={`w-8 h-8 ${stat.color}`} />
                </div>
                <div className="text-3xl font-bold text-gray-900 mb-2">
                  {stat.value}
                </div>
                <div className="text-gray-800 font-semibold mb-1">
                  {stat.label}
                </div>
                <div className="text-sm text-gray-500">
                  {stat.description}
                </div>
              </div>
            );
          })}
        </div>

        <div className="mt-12 text-center bg-white p-8 rounded-xl shadow-lg">
          <h3 className="text-xl font-bold text-gray-900 mb-4">
            Why Invest in Our Platform?
          </h3>
          <div className="grid md:grid-cols-3 gap-6 text-left">
            <div>
              <h4 className="font-semibold text-gray-800 mb-2">Proven Track Record</h4>
              <p className="text-sm text-gray-600">
                Consistent growth in user adoption and successful project completion rates
              </p>
            </div>
            <div>
              <h4 className="font-semibold text-gray-800 mb-2">Scalable Impact</h4>
              <p className="text-sm text-gray-600">
                Each investment multiplies across our network, creating exponential value
              </p>
            </div>
            <div>
              <h4 className="font-semibold text-gray-800 mb-2">Sustainable Growth</h4>
              <p className="text-sm text-gray-600">
                Building long-term economic development through SME empowerment
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};

export { StatsSection };
export default StatsSection;