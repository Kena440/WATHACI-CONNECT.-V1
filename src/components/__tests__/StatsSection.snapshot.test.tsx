import { render } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import StatsSection from '../StatsSection';

vi.mock('@/lib/supabase', () => {
  const from = (table: string) => {
    if (table === 'business_stats') {
      return {
        select: () => ({
          eq: () => ({
            order: async () => ({ data: [], error: null }),
          }),
        }),
      };
    }
    if (table === 'freelancers') {
      return {
        select: () => Promise.resolve({ count: 0 }),
      };
    }
    if (table === 'profiles') {
      return {
        select: () => ({
          eq: () => Promise.resolve({ count: 0 }),
        }),
      };
    }
    return {
      select: () => Promise.resolve({ data: [], error: null }),
    };
  };
  return { supabase: { from } };
});

describe('StatsSection', () => {
  it('matches snapshot', () => {
    const { container } = render(<StatsSection />);
    expect(container).toMatchSnapshot();
  });
});
