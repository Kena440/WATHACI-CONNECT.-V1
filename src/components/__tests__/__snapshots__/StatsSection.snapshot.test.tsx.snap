// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StatsSection > matches snapshot 1`] = `
<div>
  <section
    class="py-16 bg-gradient-to-r from-orange-50 to-green-50"
  >
    <div
      class="max-w-7xl mx-auto px-6"
    >
      <div
        class="text-center mb-12"
      >
        <h2
          class="text-3xl font-bold text-gray-900 mb-4"
        >
          Our Impact & Growth
        </h2>
        <p
          class="text-lg text-gray-600 max-w-3xl mx-auto"
        >
          Real impact metrics that demonstrate our platform's success in empowering SMEs, creating jobs, and driving economic growth across Africa
        </p>
      </div>
      <div
        class="grid md:grid-cols-4 gap-6"
      >
        <div
          class="text-center bg-white p-6 rounded-xl shadow-lg hover:shadow-xl transition-all duration-300 hover:scale-105"
        >
          <div
            class="inline-flex items-center justify-center w-16 h-16 rounded-full bg-gray-100 mb-4"
          >
            <svg
              class="lucide lucide-dollar-sign w-8 h-8 text-green-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="12"
                x2="12"
                y1="2"
                y2="22"
              />
              <path
                d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
              />
            </svg>
          </div>
          <div
            class="text-3xl font-bold text-gray-900 mb-2"
          >
            ...
          </div>
          <div
            class="text-gray-800 font-semibold mb-1"
          >
            Total Funding Raised
          </div>
          <div
            class="text-sm text-gray-500"
          >
            Capital mobilized for SME growth
          </div>
        </div>
        <div
          class="text-center bg-white p-6 rounded-xl shadow-lg hover:shadow-xl transition-all duration-300 hover:scale-105"
        >
          <div
            class="inline-flex items-center justify-center w-16 h-16 rounded-full bg-gray-100 mb-4"
          >
            <svg
              class="lucide lucide-target w-8 h-8 text-blue-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <circle
                cx="12"
                cy="12"
                r="6"
              />
              <circle
                cx="12"
                cy="12"
                r="2"
              />
            </svg>
          </div>
          <div
            class="text-3xl font-bold text-gray-900 mb-2"
          >
            ...
          </div>
          <div
            class="text-gray-800 font-semibold mb-1"
          >
            Projects Completed
          </div>
          <div
            class="text-sm text-gray-500"
          >
            Successful business initiatives
          </div>
        </div>
        <div
          class="text-center bg-white p-6 rounded-xl shadow-lg hover:shadow-xl transition-all duration-300 hover:scale-105"
        >
          <div
            class="inline-flex items-center justify-center w-16 h-16 rounded-full bg-gray-100 mb-4"
          >
            <svg
              class="lucide lucide-users w-8 h-8 text-purple-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
              />
              <circle
                cx="9"
                cy="7"
                r="4"
              />
              <path
                d="M22 21v-2a4 4 0 0 0-3-3.87"
              />
              <path
                d="M16 3.13a4 4 0 0 1 0 7.75"
              />
            </svg>
          </div>
          <div
            class="text-3xl font-bold text-gray-900 mb-2"
          >
            ...
          </div>
          <div
            class="text-gray-800 font-semibold mb-1"
          >
            Jobs Created
          </div>
          <div
            class="text-sm text-gray-500"
          >
            Employment opportunities generated
          </div>
        </div>
        <div
          class="text-center bg-white p-6 rounded-xl shadow-lg hover:shadow-xl transition-all duration-300 hover:scale-105"
        >
          <div
            class="inline-flex items-center justify-center w-16 h-16 rounded-full bg-gray-100 mb-4"
          >
            <svg
              class="lucide lucide-building2 w-8 h-8 text-orange-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 22V4a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v18Z"
              />
              <path
                d="M6 12H4a2 2 0 0 0-2 2v6a2 2 0 0 0 2 2h2"
              />
              <path
                d="M18 9h2a2 2 0 0 1 2 2v9a2 2 0 0 1-2 2h-2"
              />
              <path
                d="M10 6h4"
              />
              <path
                d="M10 10h4"
              />
              <path
                d="M10 14h4"
              />
              <path
                d="M10 18h4"
              />
            </svg>
          </div>
          <div
            class="text-3xl font-bold text-gray-900 mb-2"
          >
            ...
          </div>
          <div
            class="text-gray-800 font-semibold mb-1"
          >
            SMEs Supported
          </div>
          <div
            class="text-sm text-gray-500"
          >
            Small businesses empowered
          </div>
        </div>
        <div
          class="text-center bg-white p-6 rounded-xl shadow-lg hover:shadow-xl transition-all duration-300 hover:scale-105"
        >
          <div
            class="inline-flex items-center justify-center w-16 h-16 rounded-full bg-gray-100 mb-4"
          >
            <svg
              class="lucide lucide-trending-up w-8 h-8 text-indigo-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="22 7 13.5 15.5 8.5 10.5 2 17"
              />
              <polyline
                points="16 7 22 7 22 13"
              />
            </svg>
          </div>
          <div
            class="text-3xl font-bold text-gray-900 mb-2"
          >
            ...
          </div>
          <div
            class="text-gray-800 font-semibold mb-1"
          >
            Active Investors
          </div>
          <div
            class="text-sm text-gray-500"
          >
            Funding partners engaged
          </div>
        </div>
        <div
          class="text-center bg-white p-6 rounded-xl shadow-lg hover:shadow-xl transition-all duration-300 hover:scale-105"
        >
          <div
            class="inline-flex items-center justify-center w-16 h-16 rounded-full bg-gray-100 mb-4"
          >
            <svg
              class="lucide lucide-heart w-8 h-8 text-red-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
              />
            </svg>
          </div>
          <div
            class="text-3xl font-bold text-gray-900 mb-2"
          >
            ...
          </div>
          <div
            class="text-gray-800 font-semibold mb-1"
          >
            Donors & Supporters
          </div>
          <div
            class="text-sm text-gray-500"
          >
            Community champions
          </div>
        </div>
        <div
          class="text-center bg-white p-6 rounded-xl shadow-lg hover:shadow-xl transition-all duration-300 hover:scale-105"
        >
          <div
            class="inline-flex items-center justify-center w-16 h-16 rounded-full bg-gray-100 mb-4"
          >
            <svg
              class="lucide lucide-award w-8 h-8 text-yellow-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m15.477 12.89 1.515 8.526a.5.5 0 0 1-.81.47l-3.58-2.687a1 1 0 0 0-1.197 0l-3.586 2.686a.5.5 0 0 1-.81-.469l1.514-8.526"
              />
              <circle
                cx="12"
                cy="8"
                r="6"
              />
            </svg>
          </div>
          <div
            class="text-3xl font-bold text-gray-900 mb-2"
          >
            ...
          </div>
          <div
            class="text-gray-800 font-semibold mb-1"
          >
            Success Stories
          </div>
          <div
            class="text-sm text-gray-500"
          >
            Businesses thriving
          </div>
        </div>
        <div
          class="text-center bg-white p-6 rounded-xl shadow-lg hover:shadow-xl transition-all duration-300 hover:scale-105"
        >
          <div
            class="inline-flex items-center justify-center w-16 h-16 rounded-full bg-gray-100 mb-4"
          >
            <svg
              class="lucide lucide-globe w-8 h-8 text-teal-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <path
                d="M12 2a14.5 14.5 0 0 0 0 20 14.5 14.5 0 0 0 0-20"
              />
              <path
                d="M2 12h20"
              />
            </svg>
          </div>
          <div
            class="text-3xl font-bold text-gray-900 mb-2"
          >
            ...
          </div>
          <div
            class="text-gray-800 font-semibold mb-1"
          >
            Countries Served
          </div>
          <div
            class="text-sm text-gray-500"
          >
            Regional impact reach
          </div>
        </div>
      </div>
      <div
        class="mt-12 text-center bg-white p-8 rounded-xl shadow-lg"
      >
        <h3
          class="text-xl font-bold text-gray-900 mb-4"
        >
          Why Invest in Our Platform?
        </h3>
        <div
          class="grid md:grid-cols-3 gap-6 text-left"
        >
          <div>
            <h4
              class="font-semibold text-gray-800 mb-2"
            >
              Proven Track Record
            </h4>
            <p
              class="text-sm text-gray-600"
            >
              Consistent growth in user adoption and successful project completion rates
            </p>
          </div>
          <div>
            <h4
              class="font-semibold text-gray-800 mb-2"
            >
              Scalable Impact
            </h4>
            <p
              class="text-sm text-gray-600"
            >
              Each investment multiplies across our network, creating exponential value
            </p>
          </div>
          <div>
            <h4
              class="font-semibold text-gray-800 mb-2"
            >
              Sustainable Growth
            </h4>
            <p
              class="text-sm text-gray-600"
            >
              Building long-term economic development through SME empowerment
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
